### PR TITLE
Code reorganization and bug fixes

### PR DESCRIPTION
**Pull request by UkcaGreen**
Code reorganization and bug fixes

### CODE REVIEW

File: practice-app/scholar_util.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def get(self):
 
         name = parser.parse_args().get('name')
         _range = parser.parse_args().get('range')
+        _range = int(_range)
```
Comment (EnesToptas) [inline practice-app/scholar_util.py:43]: Since range is not required, parse_args might return none and casting to integer might fail.

Comment (UkcaGreen) [reply practice-app/scholar_util.py]: I handled the None input case and other potentially problematic inputs.

**Comment by UkcaGreen** [comment]
Range argument is handled as following. 

It is expected to be an integer.

If the range argument:
1. is a numerical value casted to integer.
2. is None, endpoint returns all publications of the author.
3. is greater than number of publications of the author, it returns all publications of the author.
4. is invalid (ie. string), it returns {"message": "Invalid range argument."}

**Comment by EnesToptas** [comment]
Seems great now, thanks.

**Pull request merged**